<launch>

  <!--
  <node pkg="usb_cam" type="usb_cam_node" name="left_camera" ns="stereo" output="screen">
    <param name="video_device" value="/dev/video2"/>
    <param name="image_width" value="1280"/>
    <param name="image_height" value="720"/>
    <param name="camera_frame_id" value="/left_camera_link"/>
    <param name="framerate" value="30"/>
    <param name="camera_name" value="left_camera"/>
    <param name="camera_info_url" type="string" value="file://$(find vision)/calibrations/left.yml" />
  </node>

  <node pkg="usb_cam" type="usb_cam_node" name="right_camera" ns="stereo" output="screen">
    <param name="video_device" value="/dev/video1"/>
    <param name="image_width" value="1280"/>
    <param name="image_height" value="720"/>
    <param name="camera_frame_id" value="/right_camera_link"/>
    <param name="framerate" value="30"/>
    <param name="camera_name" value="right_camera"/>
    <param name="camera_info_url" type="string" value="file://$(find vision)/calibrations/right.yml" />
  </node>

  <node pkg="image_view" type="image_view" name="left_viewer" output="screen">
    <remap from="/image" to="/stereo/left_camera/image_raw"/>
  </node>
cd
  <node pkg="image_view" type="image_view" name="right_viewer" output="screen">
    <remap from="/image" to="/stereo/right_camera/image_raw"/>
  </node>
-->

  <node pkg="vision" type="vision_node" name="vision" output="screen">
    <param name="loop_rate" value="30"/>
    <param name="white_threshold" value="200"/>
    <param name="base_frame_id" value="/base_link"/>
    <param name="num_cameras" value="2"/>
    <param name="resolution" value="0.05"/>
    <param name="x_min" value="0.0"/>
    <param name="x_max" value="2.0"/>
    <param name="y_min" value="-0.75"/>
    <param name="y_max" value="0.75"/>

    <remap from="/image_0" to="/left_camera/image_raw"/>
    <remap from="/camera_info_0" to="/left_camera/camera_info"/>
    <param name="/camera_0_frame_id" value="/left_camera_link"/>

    <remap from="/image_1" to="/right_camera/image_raw"/>
    <remap from="/camera_info_1" to="/right_camera/camera_info"/>
    <param name="/camera_1_frame_id" value="/right_camera_link"/>
  </node>

</launch>
